support/k8s/k8s-workload-registrar/mode-crd/controllers/pod_controller.go

```diff
@@ -16,46 +16,87 @@ limitations under the License.
 package controllers
 
 import (
+	"bytes"
 	"context"
+	"errors"
+	"strings"
+	"text/template"
 
 	"github.com/sirupsen/logrus"
 	federation "github.com/spiffe/spire/support/k8s/k8s-workload-registrar/federation"
 	spiffeidv1beta1 "github.com/spiffe/spire/support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1"
 
 	corev1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	PodNameIDLabel           string = "pod_name"
+	PodUIDLabel              string = "pod_uid"
+	NamespaceIDLabel         string = "namespace"
+	PodServiceAccountIDLabel string = "service_account"
+	PodHostnameLabel         string = "hostname"
+	PodNodeNameLabel         string = "node_name"
+)
+
 // PodReconcilerConfig holds the config passed in when creating the reconciler
 type PodReconcilerConfig struct {
-	Client             client.Client
-	Cluster            string
-	Ctx                context.Context
-	DisabledNamespaces []string
-	Log                logrus.FieldLogger
-	PodLabel           string
-	PodAnnotation      string
-	Scheme             *runtime.Scheme
-	TrustDomain        string
+	Client                client.Client
+	Cluster               string
+	Ctx                   context.Context
+	DisabledNamespaces    []string
+	Log                   logrus.FieldLogger
+	PodLabel              string
+	PodAnnotation         string
+	Scheme                *runtime.Scheme
+	TrustDomain           string
+	IdentityTemplate      string
+	Context               map[string]string
+	IdentityTemplateLabel string
```

Comment:
nit: Move `IdentityTemplateLabel` to be under `IdentityTemplate` to match the struct in config_crd.go